pandas/tests/extension/decimal/array.py

```diff
@@ -69,6 +70,7 @@ def dtype(self):
 
     @classmethod
     def _from_sequence(cls, scalars, dtype=None, copy=False):
+        scalars = [x if not isna(x) else DecimalDtype.na_value for x in scalars]
```

Comment:
The test for a decimals was failing because a NA value leads to a `np.nan` in the aggregated results due to this line: https://github.com/pandas-dev/pandas/blob/master/pandas/core/groupby/groupby.py#L1702.

This causes the `_from_sequence` constructor to fail because all scalars aren't the same type, then the results fall back to creating a series of objects. There are other ways to fix this, but this seemed the least intrusive.

Reply:
i would rather not backport this as this is not a robust soln. pls remove this and below and just xfail those tests (json/decimal). you can open an issue to address in 1.3.

Reply:
in groupby L1702 it should should probably return a dtype-specific NA

Reply:
Ok sounds good, I'll do the above and open a new issue to try and fix there